canarytokens/aws_infra/plan_generation.py

```diff
@@ -40,160 +78,89 @@ def generate_tf_variables(canarydrop: Canarydrop, plan):
             tf_variables["s3_objects"].append(
                 {
-                    "bucket": bucket["bucket_name"],
-                    "key": s3_object["object_path"],
+                    "bucket": bucket[AssetLabel.BUCKET_NAME],
+                    "key": s3_object,
                     "content": "".join(
-                        [
-                            random.choice(string.ascii_letters + string.digits)
-                            for _ in range(random.randint(5, 1000))
-                        ]
+                        random.choice(
+                            string.ascii_letters + string.digits,
+                            k=random.randint(5, 1000),
+                        )
                     ),
                 }
             )
     return tf_variables
 
 
-def generate_s3_bucket():
+async def _add_assets_for_type(
+    asset_type: AWSInfraAssetType,
+    aws_deployed_assets: dict,
+    aws_inventoried_assets: dict,
+    plan: dict,
+):
     """
-    Return a name for a S3 bucket.
+    Add assets of a specific type to the plan.
     """
-    separator = random.choice(["", "-"])
-    suffix = "".join(
-        [random.choice(string.ascii_lowercase + string.digits) for _ in range(10)]
+    count = _get_decoy_asset_count(
+        aws_inventoried_assets,
+        asset_type,
+        aws_deployed_assets,
     )
-    return f"{separator.join([random.choice(s) for s in [NAME_ENVS, NAME_TARGETS]])}{separator}{suffix}"
-
+    if count <= 0:
+        return
 
-def generate_s3_object():
-    """
-    Return a path for a S3 object.
-    """
-    objects = ["object", "data", "text", "passwords"]
-    directory = "".join(
-        [random.choice(string.ascii_letters + string.digits) for _ in range(10)]
+    asset_names = await data_generation.generate_names(
+        asset_type, aws_inventoried_assets.get(asset_type, []), count
     )
-    return f"{random.randint(2000, 2025)}/{directory}/{random.choice(objects)}"
-
-
-def generate_sqs_queue():
-    """
-    Return a name for a SQS queue.
-    """
-    separator = random.choice(["", "-", "_"])
 
-    return f"{separator.join([random.choice(s) for s in [NAME_ENVS, NAME_TARGETS]])}{separator}"
-
-
-def generate_ssm_parameter():
-    separator = random.choice(["", "-", "_"])
-    return f"{separator.join([random.choice(s) for s in [NAME_ENVS, NAME_TARGETS]])}{separator}"
-
-
-def generate_secretsmanager_secret():
-    """
-    Return a name for a Secrets Manager secret.
-    """
-    separator = random.choice(["", "-", "_", "+", "=", "@"])
-    return f"{separator.join([random.choice(s) for s in [NAME_ENVS, NAME_TARGETS]])}{separator}"
-
-
-def generate_dynamo_table():
-    """
-    Return a name for a DynamoDB table.
-    """
-    separator = random.choice(["", "-", "_"])
-    return f"{separator.join([random.choice(s) for s in [NAME_ENVS, NAME_TARGETS]])}{separator}"
+    config = _ASSET_TYPE_CONFIG[asset_type]
+    asset_name_key = config["asset_key"]
```

Comment:
This code seems odd ... don't we already have `asset_type` from which can later derive `asset_name_key` later only when needed? (Perhaps with `str(asset_type)` if that's the enum.)

```suggestion
    config = _ASSET_TYPE_CONFIG[asset_type]
```